edgelet/edgelet-kube/src/runtime.rs

```diff
@@ -236,11 +237,39 @@ where
     }
 
     fn system_info(&self) -> Self::SystemInfoFuture {
-        // TODO: Implement this.
-        Box::new(future::ok(SystemInfo::new(
-            "linux".to_string(),
-            "x86_64".to_string(),
-        )))
+        let result = self
+            .client
+            .lock()
+            .expect("Unexpected lock error")
+            .borrow_mut()
+            .list_nodes()
+            .map_err(|err| {
+                Error::from(err.context(ErrorKind::RuntimeOperation(RuntimeOperation::ListNodes)))
+            })
+            .map(|nodes| {
+                // Accumulate the architectures and their node counts into a hashmap
+                let mut arch_map: HashMap<String, u32> = HashMap::new();
+                for node in nodes.items.into_iter() {
+                    *(arch_map
+                        .entry(node.status.unwrap().node_info.unwrap().architecture)
+                        .or_insert(1)) += 1;
+                }
+
+                // Flatten the hashmap into a formatted string
+                let mut arch_string = String::from("[\n");
```

Comment:
Making the string as JSON array does not seem like the nicest solution. Initially, I had an idea  we add `SystemInfo` type that will live in `edgelete-kube` only. As opposite to `SystemInfo` in `edgelet-core` it will have collection `architectures: Vec<Architecture>`.

But what I figured out that this approach is rather complicated that it should be. I can describe what approach I have in mind in person if you are interested.

For now we can just all a new struct `Architecture` compose a list of architectures and serialize it to a json string.

```
#[derive(Debug, serde_derive::Serialize)]
pub struct Architecture{
    name: String,
    nodes_count: u32
}
```